packages/pangraph/src/reconsensus/remove_nodes.rs

```diff
@@ -47,10 +74,19 @@ fn remove_nodes_from_graph(graph: &mut Pangraph, node_ids: &[NodeId]) {
     // remove from node dictionary
     graph.nodes.remove(&node_id);
 
-    // remove from path
+    // === old implementation. Assumes only one node to remove per path ===
+    // === but there currently is an edge case where multiple empty nodes with the same id could be present ===
+    // // remove from path
+    // let path_nodes = &mut graph.paths.get_mut(&path_id).unwrap().nodes;
+    // let node_idx = path_nodes.iter().position(|&n| n == node_id).unwrap();
+    // path_nodes.remove(node_idx);
+
+    //  remove all nodes with the same id
     let path_nodes = &mut graph.paths.get_mut(&path_id).unwrap().nodes;
-    let node_idx = path_nodes.iter().position(|&n| n == node_id).unwrap();
-    path_nodes.remove(node_idx);
+    // while there is a node with this id, remove it:
+    while let Some(node_idx) = path_nodes.iter().position(|&n| n == node_id) {
```

Comment:
This is the second fix: sometimes a path could have more than one empty node with the same id. The same id happens only if the nodes are adjacent. This removes all of them.

A more elegant solution might be to not produce these empty nodes at all (modify the merging procedure?) or to make sure they are generated with different id.